modules/fedimint-lnv2-client/src/lib.rs

```diff
@@ -375,6 +375,34 @@ impl LightningClientModule {
         Ok(operation_id)
     }
 
+    async fn get_next_operation_id(&self, invoice: &Bolt11Invoice) -> anyhow::Result<OperationId> {
+        for payment_attempt in 0..u64::MAX {
+            let operation_id = OperationId::from_encodable((invoice.clone(), payment_attempt));
+
+            if !self.client_ctx.operation_exists(operation_id).await {
+                return Ok(operation_id);
+            }
+
+            if self.client_ctx.has_active_states(operation_id).await {
+                bail!("Previous payment attempt still in progress. Previous Operation Id: {operation_id}");
+            }
+
+            let mut stream = self
+                .subscribe_send(operation_id)
+                .await
+                .expect("operation_id exists")
+                .into_stream();
+
+            while let Some(state) = stream.next().await {
+                if let SendState::Success(..) = state {
+                    bail!("Previous payment attempt was successful. Previous Operation Id: {operation_id}");
+                }
+            }
```

Comment:
nit: maybe leave a comment here why this won't potentially block for a long time. We can only assume that because we checked for active states and there were none, so by definition a final state has to have been reached.